src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -455,22 +455,39 @@ private void CreateGenericContentFromTemplate(Guid releaseId, Release newRelease
                 });
         }
 
-        public async Task<Either<ActionResult, Unit>> RemoveDataFilesAsync(Guid releaseId, Guid fileId)
+        public async Task<Either<ActionResult, Unit>> RemoveDataFiles(Guid releaseId,
+            Guid fileId,
+            bool removingReplacedSubject = false)
         {
             return await _persistenceHelper
                 .CheckEntityExists<Release>(releaseId)
                 .OnSuccess(_userService.CheckCanUpdateRelease)
                 .OnSuccess(() => CheckReleaseFileReferenceExists(fileId))
-                .OnSuccess(releaseFileReference => CheckCanDeleteDataFiles(releaseId, releaseFileReference))
-                .OnSuccess(_ => GetDeleteDataFilePlan(releaseId, fileId))
-                .OnSuccess(async deletePlan =>
+                .OnSuccess(releaseFileReference =>
                 {
-                    await _dataBlockService.DeleteDataBlocks(deletePlan.DeleteDataBlockPlan);
-                    await _releaseSubjectService.SoftDeleteSubjectOrBreakReleaseLink(releaseId, deletePlan.SubjectId);
-
-                    return await _releaseFilesService
-                        .DeleteDataFiles(releaseId, fileId)
-                        .OnSuccessVoid(async () => await RemoveFileImportEntryIfOrphaned(deletePlan));
+                    return CheckCanDeleteDataFiles(releaseId, releaseFileReference)
+                        .OnSuccess(_ => GetDeleteDataFilePlan(releaseId, fileId))
+                        .OnSuccess(async deletePlan =>
+                        {
+                            // Unless removing a Subject after completing a replacement
+                            // Delete any replacement Subject that might exist
+                            if (!removingReplacedSubject)
```

Comment:
Why use a boolean flag? Wouldn't it be better to check that there the `ReplacedById` is not null to remove the file?

Reply:
I think I understand what you are saying and that is already the case inside the `if`. It's elaborated over two lines to name it `replacementInProgress` to make it a bit easier to understand.

```
var replacementInProgress = releaseFileReference.ReplacedById;
if (replacementInProgress.HasValue)
{
    await RemoveDataFiles(releaseId, replacementInProgress.Value);
}
```

I didn't want to use a boolean flag but the purpose of that is to prevent deleting the replacmeent when a replacement is completing and the original gets removed, since that delegates to this method to remove the original.

Until the original is removed, the original file has a `replacedBy` and the replacement file has a `replacing`, both referencing each other. We need a way to not remove the replacement here otherwise neither files will exist.

Reply:
Ah think I understand the motivation. Might I suggest updating the original file reference to null in the `ReplacementService` before calling `RemoveDataFiles`? This way we can keep this method clean and avoid having the flag.

Reply:
Cool, good suggestion, thanks. I've done this now in method `ReplacementService.RemoveOriginalSubjectAndFileFromRelease`